### PR TITLE
BUG: setitem into td64/dt64 series/frame with Categorical[strings]

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -136,6 +137,22 @@
...
+    dtype = value.dtype
+
+    if isinstance(dtype, np.dtype):
+        return (
+            dtype == np.dtype("object")
+            and lib.infer_dtype(value, skipna=False) == "string"
+        )
+    elif isinstance(dtype, CategoricalDtype):
+        return dtype.categories.inferred_type == "string"
+    return dtype == "string"
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py:153]: what hits here, other EAs? but not numpy types e.g. `np.str` would be above, right?

maybe a comment to this effect

File: pandas/core/arrays/datetimelike.py [hunk 2]
```diff
@@ -136,6 +137,22 @@
...
 
 
+def is_all_strings(value: ArrayLike) -> bool:
+    """
+    Check if this is an array of strings that we should try parsing.
+    """
+    dtype = value.dtype
+
+    if isinstance(dtype, np.dtype):
+        return (
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py:147]: shoudn't this be in core.dtypes.common?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
this a user facing bug?

**Comment by jbrockmendel** [comment]
> this a user facing bug?

yes, updated with whatsnew

File: pandas/core/dtypes/common.py [hunk 3]
```diff
@@ -1788,3 +1789,23 @@ def pandas_dtype(dtype) -> DtypeObj:
         raise TypeError(f"dtype '{dtype}' not understood")
 
     return npdtype
+
+
+def is_all_strings(value: ArrayLike) -> bool:
```
Comment (jreback) [inline pandas/core/dtypes/common.py:1794]: wouldn't hurt to add some unit tests for this when you have a chance :->

**Review by jreback — APPROVED** [review]

**Pull request merged**